Program.cs

```diff
@@ -1,100 +1,116 @@
+                        }
+                        else
+                        {
+                            Console.WriteLine(options + " is not a valid option!");
+                        }
+                        break;
 
                     case 2:
-                    Console.WriteLine("\n");
-                    Console.WriteLine($"Today is {DateTime.Today:D}. It is a beautiful {DateTime.Now.DayOfWeek}." + "\n");
-                    break;
+                        Console.WriteLine($"\nToday is {DateTime.Today:D}. It is a beautiful {DateTime.Now.DayOfWeek}.\n");
+                        break;
 
                     case 3:
-                    Console.WriteLine("\n");
-                    // Requesting the user's input.
-                    Console.WriteLine("Please enter a number.");
-                    // The data is stored in "num1".
-                    int num1 = Convert.ToInt32(Console.ReadLine());
-                    // Requesting the user's input.
-                    Console.WriteLine("Please enter another number.");
-                    // The data is stored in "num1".
-                    int num2 = Convert.ToInt32(Console.ReadLine());
-                    // Requesting the user's input.
-                    Console.WriteLine("Please enter an arithmetic operator, either addition or subtraction.");
-                    // The operator is stored in "arithmetic"".
-                    var arithmetic = Console.ReadLine();
-                    Console.WriteLine("\n");
+                        // Requesting the user's input for the first number.
+                        Console.WriteLine("\nPlease enter a number.");
+                        // The data is stored in "num1".
+                        if (!int.TryParse(Console.ReadLine(), out int num1))
+                        {
+                            Console.WriteLine("Invalid input. Read please!");
+                            break;
+                        }
+
+                        // Requesting the user's input for the second number.
+                        Console.WriteLine("Please enter another number.");
+                        // The data is stored in "num2".
+                        if (!int.TryParse(Console.ReadLine(), out int num2))
+                        {
+                            Console.WriteLine("Invalid input. Read please!");
+                            break;
+                        }
 
-                    // Addition.
-                    int add = num1 + num2;
-                    // Subtraction
-                    int sub = num1 - num2;
+                        // Requesting the user's input for the arithmetic operator.
+                        Console.WriteLine("Please enter an arithmetic operator, either + for addition or - for subtraction.");
+                        // The operator is stored in "arithmetic".
+                        string arithmetic = Console.ReadLine();
```

Comment:
where is the check to make sure the user entered a correct input

Reply:
It's in the if statement below.